projects/packages/jetpack-mu-wpcom/src/features/calypso-locale-sync/sync-wp-admin-to-calypso.php

```diff
@@ -27,32 +30,42 @@ function update_calypso_locale( $locale ) {
 
 /**
  * Sync locale updated via /wp-admin/profile.php to Calypso.
- *
- * @param array   $meta   Meta values and keys for the user.
- * @param WP_User $user   User object.
- * @param boolean $update Whether the user is being updated rather than created.
  */
-function sync_wp_admin_locale_to_calypso( $meta, $user, $update ) {
-	$locale            = $meta['locale'];
-	$old_locale        = get_user_locale( $user );
-	$is_user_connected = ( new Connection_Manager( 'jetpack' ) )->is_user_connected();
+function sync_wp_admin_locale_on_profile_update() {
+	$locale = null;
+
+	if ( isset( $_POST['locale'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$locale = sanitize_text_field( wp_unslash( $_POST['locale'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+	}
+
+	if ( '' === $locale ) {
+		$locale = 'en';
+	}
```

Comment:
Just curious, was the lack of this handling the underlying issue that caused many calls to `update_calypso_locale`?

Asking because the post-mortem P2 post mentions that "_This function ran on insert_user_meta action without examining which usermeta was changing_", but I think the function was already doing that before (it was bailing if `$old_locale === $locale`).

So I guess we had cases where `$locale` didn't have the value we expected, which is what we are handling now.

Reply:
It wasn't that check because the param is empty only with the hook `personal_options_update`, but when using `insert_user_meta`, the value received for English is `en_US`.

The case that caused the issue was in [this condition](https://github.com/Automattic/jetpack/blob/f46c68670dba6b37f8d568152f4a4008c298539e/projects/packages/jetpack-mu-wpcom/src/features/calypso-locale-sync/sync-wp-admin-to-calypso.php#L40):

```
if ( ! $update || $old_locale === **$locale** || ! $is_user_connected ) {
  // Only proceed for locale changes on an existing connected WPCOM user.
  return $meta;
}
```

`$locale` from `$meta['locale']` is an empty string for Site Default when using the hook `insert_user_meta`. I debugged it to confirm it.

That condition needed a check to also bail when the user has selected Site Default and the site locale (WPLANG) was already synced on Calypso.